VenafiTppPS/Code/Public/Get-TppPermission.ps1

```diff
@@ -1,222 +1,218 @@
+
+.LINK
+https://docs.venafi.com/Docs/18.2SDK/TopNav/Content/SDK/WebSDK/API_Reference/r-SDK-GET-Permissions-object-guid-principal.php?tocpath=REST%20API%20reference%7CPermissions%20programming%20interfaces%7C_____5
+
+#>
+function Get-TppPermission {
+
+    [CmdletBinding(DefaultParameterSetName = 'ByObject')]
+    param (
+        [Parameter(Mandatory, ParameterSetName = 'ByObject', ValueFromPipeline)]
+        [TppObject] $InputObject,
+
+        [Parameter(Mandatory, ParameterSetName = 'ByPath', ValueFromPipeline)]
+        [ValidateNotNullOrEmpty()]
+        [ValidateScript( {
+                if ( $_ | Test-TppDnPath ) {
+                    $true
+                } else {
+                    throw "'$_' is not a valid DN path"
+                }
+            })]
+        [Alias('DN', 'CertificateDN')]
+        [String[]] $Path,
+
+        [Parameter(Mandatory, ParameterSetName = 'ByGuid', ValueFromPipeline)]
+        [ValidateNotNullOrEmpty()]
+        [Guid[]] $Guid,
+
+        [Parameter()]
+        [ValidateScript( {
+                $_ -match '(AD|LDAP)+\S+:\w{32}$' -or $_ -match 'local:\w{8}-\w{4}-\w{4}-\w{4}-\w{12}$'
+            })]
+        [Alias('PrefixedUniversal')]
+        [string[]] $PrefixedUniversalId,
+
+        [Parameter()]
+        [Alias('ExplicitImplicit')]
+        [switch] $Explicit,
+
+        [Parameter()]
+        [ValidateSet('Group Membership', 'Name', 'Internet Email Address', 'Given Name', 'Surname')]
+        [string[]] $Attribute,
+
+        [Parameter()]
+        [TppSession] $TppSession = $Script:TppSession
+    )
+
+    begin {
+        $TppSession.Validate()
+
+        Write-Verbose ("Parameter set {0}" -f $PsCmdlet.ParameterSetName)
+
+        $params = @{
+            TppSession = $TppSession
+            Method     = 'Get'
+            UriLeaf    = 'placeholder'
+        }
+
+        $returnObject = @()
+    }
+
+    process {
+
+        if ( $PSBoundParameters.ContainsKey('Path') ) {
+            $InputObject = Get-TppObject -Path $Path -TppSession $TppSession
+        } elseif ( $PSBoundParameters.ContainsKey('Guid') ) {
+            $InputObject = $Guid | ConvertTo-TppPath -TppSession $TppSession | Get-TppObject -TppSession $TppSession
+        }
+
+
+        foreach ( $thisObject in $InputObject ) {
+
+            $uriBase = ('Permissions/Object/{{{0}}}' -f $thisObject.Guid)
+            $params.UriLeaf = $uriBase
+
+            if ( $PSBoundParameters.ContainsKey('PrefixedUniversalId') ) {
+                $principals = $PrefixedUniversalId
+            } else {
+                # get list of principals permissioned to this object
+                $principals = Invoke-TppRestMethod @params
+            }
+
+            foreach ( $principal in $principals ) {
+
+                $params.UriLeaf = $uriBase
```

Comment:
this was the core issue, UriLeaf needed to be reset so it wasn't continually appended to in this loop